CONTRIBUTING.md

```diff
@@ -92,34 +96,61 @@ To (re)format the Gloo source code, run:
 $ cargo fmt --all
 ```
 
-## Gloo Crate Checklist
+## Workflow
 
-Here is a checklist that all Gloo utility crates should fulfill:
+Designing APIs for Gloo, its utility crates, and interfaces between them takes a
+lot of care. The design space is large, and there is a lot of prior art to
+consider. When coming to consensus on a design, we use a simplified, informal
+version of [our RFC process][rfcs], where we have design discussions inside the
+Gloo issues tracker.
```

Comment:
Nit: It is (and should be) ambiguous when a design spec is warranted and when a direct PR is more appropriate. Should we encourage those who are unsure to open an issue?

Reply:
Yes, as a general rule issues should be opened first.

It's only after there is some consensus about the design that a PR can be opened.

Reply:
FWIW, this text is largely addressing "new designs/APIs" where I think we practically always want the design discussion first before any PRs.

For bug fixes and other semver compatible changes that don't increase the API surface, I do think we can usually jump straight to a PR, but it also isn't really addressed by this documentation. We could probably add an aside about this somewhere. I'll look into it.

Reply:
Added a note about purely internal bug fixes in the latest commit.